cmd/buildctl/build.go

```diff
@@ -203,6 +208,12 @@ func buildAction(clicontext *cli.Context) error {
 		return errors.Wrap(err, "invalid local")
 	}
 
+	// parse --opt to find the context ones that point to OCILayout stores
+	solveOpt.OCIStores, err = getOCILayoutStores(solveOpt.FrontendAttrs)
```

Comment:
You don't want to add a special flag for this? Would be more consistent if this was similar to `--local`

Reply:
Oh definitely. I originally had that, but pulled it out. I had a hard time understanding the relationship between `--local something` and `--opt context:foo=local:something`. Are they both valid? Where does the `--opt` one get added then?

Reply:
I see you added `build.ParseOCILayout` that looks correct. Why do you need to do this opt parsing at all in here now? This part is already in the Dockerfile frontend.

`--oci-layout foo=/path/to/dir --opt context:name=oci-layout://foo@sha256:abc`

`--oci-layout` is handled in buildctl (here)
`--opt context` is handled in Dockerfile frontend

Reply:
That clarifies it. It wasn't clear before.

* `--local` and `--oci-layout`: add OCI stores giving each a path and a unique identifier
* `--opt context`: reference previously identified stores for context

Is that in the docs somewhere? It is something that took me some time to grok.